### PR TITLE
Php log analyzer

### PR DESCRIPTION
**Pull request by aspaan**

### CODE REVIEW

File: app/shared/models/solution-ui-model/othersolutions.ts [hunk 1]
```diff
@@ -212,6 +212,183 @@ export class RunRemoteProfiler extends SolutionUIModelBase {
     }
 }
 
+export class UpgradeDatabase extends SolutionUIModelBase {
```
Comment (ShekharGupta1988) [inline app/shared/models/solution-ui-model/othersolutions.ts]: Maybe "Scale Up Your Clear DB Database"?

The title should be clear for customer.

File: app/shared/models/solution-ui-model/othersolutions.ts [hunk 2]
```diff
@@ -212,6 +212,183 @@ export class RunRemoteProfiler extends SolutionUIModelBase {
...
+    constructor(rank: number, _portalService: PortalService) {
+        super(rank, new SolutionProperties(), _portalService);
+
+        this.properties.id = 105;
+        this.properties.title = "Upgrade Database to a higher plan";
```
Comment (ShekharGupta1988) [inline app/shared/models/solution-ui-model/othersolutions.ts]: same here

**Comment by aspaan** [comment]
image

image

File: app/shared/models/solution-ui-model/othersolutions.ts (hunk 1 above)
Comment (ShekharGupta1988) [inline app/shared/models/solution-ui-model/othersolutions.ts]: can this also be Azure DB?

File: app/shared/models/solution-ui-model/othersolutions.ts [hunk 3]
```diff
@@ -212,6 +212,183 @@ export class RunRemoteProfiler extends SolutionUIModelBase {
...
+    run(): void {
+    }
+}
+
+export class ContactDatabaseProvider extends SolutionUIModelBase {
+    constructor(rank: number, _portalService: PortalService) {
+        super(rank, new SolutionProperties(), _portalService);
+
+        this.properties.id = 106;
+        this.properties.title = "Contact ClearDB or Azure Database for more information";
```
Comment (ShekharGupta1988) [inline app/shared/models/solution-ui-model/othersolutions.ts]: Add Support

File: app/shared/models/solution-ui-model/othersolutions.ts [hunk 4]
```diff
@@ -212,6 +212,183 @@ export class RunRemoteProfiler extends SolutionUIModelBase {
...
+        
+        this.properties.type = Enums.SolutionType.BestPractices;
```
Comment (ShekharGupta1988) [inline app/shared/models/solution-ui-model/othersolutions.ts]: Change this

File: app/shared/models/solution-ui-model/othersolutions.ts [hunk 5]
```diff
@@ -212,6 +212,183 @@ export class RunRemoteProfiler extends SolutionUIModelBase {
...
+    run(): void {
+    }
+}
+
+export class IncreasePHPTimeOUt extends SolutionUIModelBase {
+    constructor(rank: number, private _parameters: INameValuePair[][], _portalService: PortalService, private _siteService: SiteService) {
+        super(rank, new SolutionProperties(), _portalService);
+
+        this.properties.id = 107;
+        this.properties.title = "Increase PHP Time Out";
```
Comment (ShekharGupta1988) [inline app/shared/models/solution-ui-model/othersolutions.ts]: this is the attention grabbing line for customers. may be "Increase your PHP App Time out"

File: app/shared/models/solution-ui-model/othersolutions.ts [hunk 6]
```diff
@@ -212,6 +212,183 @@ export class RunRemoteProfiler extends SolutionUIModelBase {
...
+        this.properties.id = 106;
+        this.properties.title = "Contact ClearDB or Azure Database for more information";
+        
+        this.properties.type = Enums.SolutionType.BestPractices;
+
+        this.properties.additionalData = [];
+
+        let metaDataItem = new SolutionMetadata();
+        metaDataItem.type = "blog";
+        metaDataItem.message = "*Contact ClearDB* support if your are using ClearDB database.";
```
Comment (stevenernst131) [inline app/shared/models/solution-ui-model/othersolutions.ts]: Change to "Contact *ClearDB Support* if your are using a ClearDB database."

File: app/shared/models/solution-ui-model/othersolutions.ts [hunk 7]
```diff
@@ -212,6 +212,183 @@ export class RunRemoteProfiler extends SolutionUIModelBase {
...
+        metaDataItem.og_Url = "http://w2.cleardb.net/support/";
+        metaDataItem.og_Title = "Contact ClearDB";
+        this.properties.additionalData.push(metaDataItem);
+        
+        metaDataItem = new SolutionMetadata();
+        metaDataItem.type = "blog";
+        metaDataItem.message = "*Contact Microsoft* support if you are using Azure Database for MySQL";
```
Comment (stevenernst131) [inline app/shared/models/solution-ui-model/othersolutions.ts]: Change to "Contact *Microsoft Support* if you are using an Azure Database for MySQL"

Comment (stevenernst131) [inline app/shared/models/solution-ui-model/othersolutions.ts]: Change to "Contact *Microsoft Support* if you are using an Azure Database for MySQL"

Comment (stevenernst131) [inline app/shared/models/solution-ui-model/othersolutions.ts]: This also should probably open ticket blade but we can do that later.

**Comment by stevenernst131** [comment]
This is a problem we have with solutions already I believe but can you make the titles consistent in terms of capitalization? i.e. "Turn on PHP logging" should be "Turn on PHP Logging".

File: app/shared/models/solution-ui-model/othersolutions.ts (hunk 7 above)
Comment (aspaan) [inline app/shared/models/solution-ui-model/othersolutions.ts]: Done

File: app/shared/models/solution-ui-model/othersolutions.ts (hunk 6 above)
Comment (aspaan) [inline app/shared/models/solution-ui-model/othersolutions.ts]: Done

File: app/shared/models/solution-ui-model/othersolutions.ts (hunk 5 above)
Comment (aspaan) [inline app/shared/models/solution-ui-model/othersolutions.ts]: Done

File: app/shared/models/solution-ui-model/othersolutions.ts (hunk 4 above)
Comment (aspaan) [inline app/shared/models/solution-ui-model/othersolutions.ts]: Done

File: app/shared/models/solution-ui-model/othersolutions.ts (hunk 3 above)
Comment (aspaan) [inline app/shared/models/solution-ui-model/othersolutions.ts]: Done

File: app/shared/models/solution-ui-model/othersolutions.ts (hunk 2 above)
Comment (aspaan) [inline app/shared/models/solution-ui-model/othersolutions.ts]: I dont get it

**Pull request merged**